OMG.Classroom/DataAccessLayer/MapperProfile/MapperProfile.cs

```diff
@@ -0,0 +1,48 @@
+﻿using AutoMapper;
+using DataAccessLayer.Dtos.AssignmentDtos;
+using DataAccessLayer.Dtos.CourseDtos;
+using DataAccessLayer.Dtos.MessageDtos;
+using DataAccessLayer.Dtos.StudentDtos;
+using DataAccessLayer.Dtos.TeacherDtos;
+using DataAccessLayer.Dtos.UserDtos;
+using DataAccessLayer.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.MapperProfile
+{
+    public class MapperProfile : Profile
+    {
+        public MapperProfile()
+        {
+            CreateMap<AssignmentDto, Assignment>().ReverseMap();
+            CreateMap<AssignmentDtoWithId, Assignment>().ReverseMap();
+            CreateMap<AssignmentDtoWithMessages, Assignment>().ReverseMap();
+            CreateMap<AssignmentDtoWithMessagesWithId, Assignment>().ReverseMap();
+
+            CreateMap<CourseDto, Course>().ReverseMap();
+            CreateMap<CourseDtoWithId, Course>().ReverseMap();
+            CreateMap<CourseDtoWithStudentsAndAssignments, Course>().ReverseMap();
+            CreateMap<CourseDtoWithStudentsAndAssignmentsWithId, Course>().ReverseMap();
+
+            CreateMap<MessageDto, Message>().ReverseMap();
+            CreateMap<MessageDtoWithId, Message>().ReverseMap();
+
+            CreateMap<StudentDto, Student>().ReverseMap();
+            CreateMap<StudentDtoWithId, Student>().ReverseMap();
+            CreateMap<StudentDtoWithAssignments, Student>().ReverseMap();
+            CreateMap<StudentDtoWIthAssignmentsWithId, Student>().ReverseMap();
+
+            CreateMap<TeacherDto, Teacher>().ReverseMap();
+            CreateMap<TeacherDtoWithId, Teacher>().ReverseMap();
+
+            CreateMap<UserDto, User>().ReverseMap();
```

Comment:
So, in our application we won't work with just user, we will work only with Students and Teachers. User only contains the properties that both user and teacher have. So, remove user dtos and put its dtos into student and teacher dtos